packages/react-native-editor/__device-tests__/helpers/utils.js

```diff
@@ -482,30 +488,67 @@ const waitForVisible = async ( driver, elementLocator, iteration = 0 ) => {
 		// if locator is not visible, try again
 		return waitForVisible( driver, elementLocator, iteration + 1 );
 	}
+
 	return locator[ 0 ];
 };
 
+/**
+ * @param {string} driver
+ * @param {string} elementLocator
+ * @param {number} iteration - Default value is 0
+ * @return {boolean} - Returns true if element is found, false otherwise
+ */
+const isElementVisible = async ( driver, elementLocator, iteration = 0 ) => {
+	const maxIteration = 5;
+	const timeout = 1000;
+
+	if ( iteration >= maxIteration ) {
+		// Element is still not visible after waiting, return false
+		return false;
+	} else if ( iteration !== 0 ) {
+		// wait before trying to locate element again
+		await driver.sleep( timeout );
+	}
+
+	const locator = await driver.elementsByXPath( elementLocator );
+	if ( locator.length !== 1 ) {
+		// if locator is not visible, try again
+		return isElementVisible( driver, elementLocator, iteration + 1 );
+	}
+
+	return true;
+};
```

Comment:
I noticed that the logic of this function is very similar to `waitForVisible`, so I'm wondering if we could call that function here instead of duplicating the logic, wdyt?

Reply:
I'll try to see what I can do, in `waitForVisible()` an error is thrown if the element is not found causing the Appium session to be terminated (IIRC we couldn't use a try-catch because of this). Let me see if there's a workaround to that, but I do agree that the two are very similar except for the return values.

Reply:
Yep, most likely we could use `try-catch` here for checking if the element is visible.

Reply:
I updated this so that `waitForVisible()` helper method does not throw an error when the element is not found, instead it will print an error and return an empty string (so the next step in the test will still fail if the element does not exist). I think this makes sense for 2 reasons:
1. Throwing an error in a test will immediately terminate the Appium session, so the test can't continue and if there are subsequent tests in the test suite, those tests will also fail
2. Noticed for a while now that if a test fails because it couldn't find the element, it's not reported properly in the tests tab in CircleCI ([example here](https://app.circleci.com/pipelines/github/wordpress-mobile/gutenberg-mobile/17320/workflows/886d74cc-4ddf-489b-a261-90d9447fa4fa/jobs/94374/steps))

The changes were done here: https://github.com/WordPress/gutenberg/pull/40670/commits/0077156bde9c865ea3b79d9bc398e13833531451

Reply:
> I updated this so that `waitForVisible()` helper method does not throw an error when the element is not found, instead it will print an error and return an empty string (so the next step in the test will still fail if the element does not exist). I think this makes sense for 2 reasons:
> 
> 1. Throwing an error in a test will immediately terminate the Appium session, so the test can't continue and if there are subsequent tests in the test suite, those tests will also fail
> 2. Noticed for a while now that if a test fails because it couldn't find the element, it's not reported properly in the tests tab in CircleCI ([example here](https://app.circleci.com/pipelines/github/wordpress-mobile/gutenberg-mobile/17320/workflows/886d74cc-4ddf-489b-a261-90d9447fa4fa/jobs/94374/steps))
> 
> The changes were done here: [0077156](https://github.com/WordPress/gutenberg/commit/0077156bde9c865ea3b79d9bc398e13833531451)

I like the idea of removing the exception, as you mentioned, this way we'll let the test case fail with the appropriate error if an element doesn't exist 👍 .